modules/learn/pages/clusters-and-availability/xdcr-enable-crossclusterversioning.adoc

```diff
@@ -0,0 +1,65 @@
+
+[#bidirectional-replication-with-sgw-and-xdcr]
+=== Bi-directional replication with Sync Gateway and XDCR
+
+Pre-requisites for a bi-directional replication with Sync Gateway and XDCR are as follows:
+
+* Set the bucket property `enableCrossClusterVersioning` to `true` to use the `mobile=[Off | Active]` flag during the processes xref:manage:manage-xdcr/create-xdcr-replication.adoc[Create a Replication] and xref:manage:manage-xdcr/xdcr-management-overview.adoc[Manage XDCR]. Then set the `mobile=[Off | Active]` flag to `Active` for a bi-directional replication with Sync Gateway 4.0+ and XDCR.
+
+* Use Sync Gateway 4.0 or a higher version. When you're using Sync Gateway 4.0 or a higher version, Sync Gateway adds HLV information (whether or not you have actively set `enableCrossClusterVersioning` to `true` in the bucket property). The existence of the HLV information causes XDCR to maintain that information, so for one-way (unidirectional) XDCR-Sync Gateway users, using Sync Gateway 4.0 or a higher version means that there are additional metadata. You must enable the bucket property `enableCrossClusterVersioning` for all buckets in the replication topology and set the flag `mobile` to `Active` in the XDCR replication to configure bi-directional replication with Sync Gateway and XDCR.
+
+For information about XDCR Active-Active with Sync Gateway 4.0+ (or to configure bi-directional replication with Sync Gateway and XDCR), see xref:learn:clusters-and-availability/xdcr-active-active-sgw.adoc[XDCR Active-Active with Sync Gateway].
+
+IMPORTANT: This is an important limitation to the _XDCR Active-Active with Sync Gateway_ feature. If you use the user created extended attributes (xattrs) in your documents, and you have more than 10 user xattrs in a document, then you cannot use the feature _XDCR Active-Active with Sync Gateway_. This is due to an internal limitation of managing extended attributes in a document. If you try to use the feature _XDCR Active-Active with Sync Gateway_ when you have more than 10 user xattrs in your document, the XDCR replication **silently skips** replicating the document that has more than 10 user xattrs. As a result, the source and target data in the replication skipped documents will not be consistent. The only way you will know this skip occured is because the Prometheus stat **subdoc_cmd_docs_skipped** will be incremented and the document will _not_ be consistent between the target and source. So, if your documents have more than 10 user xattrs, do _not_ enable the property `enableCrossClusterVersioning` with a purpose to use the _XDCR Active-Active with Sync Gateway_ feature.
+
+[#hlv-data-maintained-in-xattr]
+== Hybrid Logical Vector (HLV) data maintained in the extended attributes
+
+The new metadata, HLV, is stored as a system user created extended attribute (xattr). The HLV metadata is also called Version Vectors. The system xattr name is `"_vv" -- "xattrs"."_vv"`. 
```

Comment:
The first sentence is a bit confusing -- it says "system user created" -- I think it should just say "system created" (otherwise, you might think, is it "system created" or "user created").

The new metadata, HLV, is stored as a system created extended attribute (xattr) called `_vv` (`xattrs._vv`).  The HLV metadata is also called Version Vectors.